python/GangaDirac/Lib/Files/DiracFile.py

```diff
@@ -661,15 +667,18 @@ def put(self, lfn='', force=False, uploadSE="", replicate=False):
                 logger.warning("LFN will be generated automatically")
                 self.lfn = ""
 
-        selfConstructedLFN = False
-
-        import glob
-        if self.remoteDir == '' and self.lfn == '':
-            import datetime
-            t = datetime.datetime.now()
-            this_date = t.strftime("%H.%M_%A_%d_%B_%Y")
-            self.lfn = os.path.join(DiracFile.diracLFNBase(), 'GangaFiles_%s' % this_date)
-            selfConstructedLFN = True
+        if self.remoteDir == '':
+            try:
+                job = self.getJobObject()
+            except AssertionError:
+                job = None
+            if job:
+                lfn_folder = os.path.join("GangaUploadedFiles", "GangaJob_%s" % job.getFQID('.'))
+            else:
+                t = datetime.datetime.now()
+                this_date = t.strftime("%H.%M_%A_%d_%B_%Y")
+                lfn_folder = os.path.join("GangaUploadedFiles", 'GangaFiles_%s' % this_date)
+            self.lfn = os.path.join(DiracFile.diracLFNBase(), lfn_folder, self.namePattern)
```

Comment:
This could simply be:
```python
if not self.remoteDir:
    try:
        job = self.getJobObject()
        lfn_folder = os.path.join("GangaUploadedFiles", "GangaJob_%s" % job.getFQID('.'))
    except AssertionError:
        t = datetime.datetime.now()
        this_date = t.strftime("%H.%M_%A_%d_%B_%Y")
        lfn_folder = os.path.join("GangaUploadedFiles", 'GangaFiles_%s' % this_date)
    self.lfn = os.path.join(DiracFile.diracLFNBase(), lfn_folder, self.namePattern)
```